### ISSUE TITLE
 bluecherry-apps build errors in ubuntu

### ISSUE THREAD

**Issue by hongson23** [bluecherrydvr/bluecherry-apps#394]
hello @antonsviridenko 
I saw that, you have a latest commits so can you please help me check issues for build bluecherry app
I cloned this repo and run scripts follow by readme note:  scripts/build_pkg_native.sh
but I have got some errors while make project under some distro ubuntu such as: 20.04LTS, 18.04LTS, 16.04 LTS 

![e1](https://user-images.githubusercontent.com/4011696/101713191-bab20b00-3ac9-11eb-9d5c-e2355d685e95.png)
![e2](https://user-images.githubusercontent.com/4011696/101713193-bc7bce80-3ac9-11eb-9116-6b48edcf8afd.png)

thanks!

**Comment by antonsviridenko** [comment]
Yes, unfortunately build process for onvif_tool build is not completed yet,

https://github.com/bluecherrydvr/onvif_tool 

you can take "onvif_tool" binary from our .deb packages and put it into the utils/ subdirectory to make the build successful.

**Comment by hongson23** [comment]
> Yes, unfortunately build process for onvif_tool build is not completed yet,
> 
> https://github.com/bluecherrydvr/onvif_tool
> 
> you can take "onvif_tool" binary from our .deb packages and put it into the utils/ subdirectory to make the build successful.

thank for your reply!
I try to download .deb packages from https://www.ubuntu.bluecherrydvr.com/pool/bionic but can not see any "onvif_tool" binary 
Can you upload this package?
thanks

**Comment by sicada** [comment]
> you can take "onvif_tool" binary from our .deb packages and put it into the utils/ subdirectory to make the build successful.

I appreciate your work, but this is a _laborious_ request to make of users. Would you consider adding the binary(ies) of onvif_tool to the github repo? Otherwise, the process of finding the right .deb package, extracting it, and locating the binary is a real pain.

I finally found the correct binary for my architecture (debian, x86_64) in this location of a bluecherry debian package:
`bluecherry_3.0.4_amd64/data/usr/lib/bluecherry/onvif_tool`

I got that package from the unstable bluecherry apt repo at:
http://unstable.bluecherrydvr.com/pool/buster/

Thanks,
-Nick

**Comment by sicada** [comment]
> thank for your reply!
> I try to download .deb packages from https://www.ubuntu.bluecherrydvr.com/pool/bionic but can not see any "onvif_tool" binary

@hongson23  That's because that repo only has bluecherry 2.x packages, and **onvif_tool** is only present in the bluecherry 3.x packages, which are currently available from unstable.bluecherrydvr.com.

Easy, right? :)

**Comment by curtishall** [comment]
> I appreciate your work, but this is a laborious request to make of user

I completely agree, this is an necessary step.  We will be adding documentation in the next few weeks on how to properly build the onvif_tool.

**Comment by sicada** [comment]
Thank you @curtishall , much appreciated!  FYI- I've been playing in depth with the bluecherry 3.x build process in docker containers and will hopefully have some suggestions for improvements there for you all in the near future. :)

**Comment by curtishall** [comment]
> I've been playing in depth with the bluecherry 3.x build process in docker containers and will hopefully have some suggestions for improvements there for you all in the near future. :)
👍 👍 

Excellent!  I've actually have been brainstorming this also.  My thoughts are docker for all supported releases (i386/amd64) and then adding a volume of the git repo and a volume for the compiled debs by distro/arch name.  I've just not had time to actually put this together in a dockerfile yet :).  If you come up with something that would be swell!
